### ISSUE TITLE
interp_prop_raiseOOM

### ISSUE THREAD

**Issue by Chobbes** [vellvm/vellvm#308]
This situation has come up a few times:

```
  interp_prop (refine_OOM_handler (F:=UBE +' DebugE +' FailureE))
    (@oom_k_spec ExternalCallE (UBE +' DebugE +' FailureE))
    (LLVM.MEM.MMEP.MMSP.MemState * (store_id * (local_env * Stack.stack * res_L1))) 
    (flip refine_res3) (raiseOOM err_msg) (raiseOOM err_msg)
```

Note: the raiseOOM trees have slightly different effect signatures.

**Comment by Chobbes** [comment]
Relatedly, this currently lives in OOMRefinementExamples.v

```
  Lemma model_undef_h_oom :
    forall {R} {E F}
      `{FailureE -< E +' F} `{UBE -< E +' F} `{OOME -< E +' F} `{OOME -< E +' PickUvalueE +' F}
      oom_msg (t' : itree (E +' F) R),
      model_undef_h eq (raiseOOM oom_msg) t' ->
      t' ≈ raiseOOM oom_msg.
```

**Comment by euisuny** [comment]
Resolved second issue (`model_undef_h_oom`) in c6d4a12 by concretizing the events into the exact events found in layer `L4`

**Comment by euisuny** [comment]
The first comment seems to be now defunct (due to updated definitions, etc.), closing issue.
